src/App.tsx

```diff
@@ -12,50 +14,93 @@ function App() {
   const [isReversed, setIsReversed] = useState<boolean>(false)
   const [leftCurrency, setLeftCurrency] = useState<string>('')
   const [rightCurrency, setRightCurrency] = useState<string>('')
+  const [exchangeRate, setExchangeRate] = useState<string|number>('')
+  const [allCurrenciesSymbols, setAllCurrenciesSymbols] = useState<Symbol[]>([])
+  const [leftSymbol, setLeftSymbol] = useState<string|undefined>()
+  const [rightSymbol, setRightSymbol] = useState<string|undefined>()
+
+  useEffect(() => {
+    addSymbolToInput({leftCurrency, rightCurrency, allCurrenciesSymbols, setLeftSymbol, setRightSymbol})
```

Comment:
First of all: don't add new features to the old rep/functional
second: your useEffect will run when you change the left or right  currency input

Reply:
+